internal/oc/release.go

```diff
@@ -148,17 +151,33 @@ func (r *release) GetMajorMinorVersion(log logrus.FieldLogger, releaseImage stri
 
 func (r *release) GetReleaseArchitecture(log logrus.FieldLogger, releaseImage string, releaseImageMirror string, pullSecret string) (string, error) {
 	var cmd string
+	var cmdMultiarch string
 	if releaseImage == "" && releaseImageMirror == "" {
 		return "", errors.New("no releaseImage nor releaseImageMirror provided")
 	}
 	if releaseImageMirror != "" {
 		cmd = fmt.Sprintf(templateImageInfo, releaseImageMirror)
+		cmdMultiarch = fmt.Sprintf(templateSkopeoDetectMultiarch, releaseImageMirror)
 	} else {
 		cmd = fmt.Sprintf(templateImageInfo, releaseImage)
+		cmdMultiarch = fmt.Sprintf(templateSkopeoDetectMultiarch, releaseImage)
 	}
-	imageInfoStr, err := execute(log, r.executer, pullSecret, cmd)
+
+	imageInfoStr, err := execute(log, r.executer, pullSecret, cmd, ocAuthArgument)
 	if err != nil {
-		return "", err
+		// TODO(WRKLDS-222) At this moment we don't have a better way to detect if the release image is a multiarch
+		//                  image. Introducing skopeo as an additional dependency, to be able to manually parse
+		//                  the manifest. https://bugzilla.redhat.com/show_bug.cgi?id=2111537 tracks the missing
+		//                  feature in oc cli.
+		skopeoImageRaw, err2 := execute(log, r.executer, pullSecret, cmdMultiarch, skopeoAuthArgument)
+		if err2 != nil {
+			return "", errors.Errorf("failed to parse image, oc: %v, skopeo: %v", err, err2)
+		}
+		_, _, _, err2 = jsonparser.Get([]byte(skopeoImageRaw), "manifests")
+		if err2 != nil {
+			return "", errors.Errorf("failed to parse image, oc: %v, json: %v", err, err2)
```

Comment:
So if we get here it means 'oc image info' failed but 'skopeo inspect' succeeded, which not sure is actually possible:) Any way, I guess it's sufficient to just return the oc err in this case (as missing 'manifests' simply means a non-multi arch image). I.e. the error message can be something like "failed to get image info using oc: %v", err).

Reply:
I think the path to reach this code is as follows - you feed `oc` with a single-arch release image and make the `oc image info` fail (for any reason); `oc` failed because it sometimes fails :) and jsonparser failed because it's a single-release image. Still you are right, printing `err2` makes no sense here because it would only tell us something like `key manifests does not exist` which does not bring much value. Changed as suggested 👍🏻